gems/sorbet-runtime/lib/types/enum.rb

```diff
@@ -185,6 +185,30 @@ def <=>(other)
     end
   end
 
+  # NB: Do not call this method. This exists to allow for a safe migration path in places where enum
+  # values are compared directly against string values.
+  #
+  # Ruby's string has a weird quirk where `'my_string' == obj` calls obj.==('my_string') if obj
+  # responds to the `to_str` method. It does not actually call `to_str` however.
+  #
+  # See https://ruby-doc.org/core-2.4.0/String.html#method-i-3D-3D
+  T::Sig::WithoutRuntime.sig {returns(String)}
+  def to_str
+    msg = 'Implicit conversion of Enum instances to strings is not allowed. Call #serialize instead.'
+    if T::Configuration.legacy_t_enum_migration_mode?
```

Comment:
I couldn't get `LegacyMigrationMode` to override this method without this check so I extracted the whole method instead. Given that it's not used for non `LegacyMigrationMode` enums it shouldn't make a difference in performance.

Reply:
The override from `LegacyMigrationMode` could only work if `include LegacyMigrationMode` comes _after_ this method definition